ucb-rit/coldfront
# Issue 571: Require completion of a survey before yearly allowance may be renewed

Fixes #571 

**Changes**

- Added a JSON field, `renewal_survey_answers` to `AllocationRenewalRequest` model to store survey responses. Default value is an empty dict.
- Introduced a new renewal form, `ProjectRenewalSurveyForm` (modeled after `SavioProjectSurveyForm`): 
  - Checks for `BRC_ONLY` or `LRC_ONLY` flag at runtime to determine which questions to display. BRC and LRC will use different survey questions.
  - Current questions are placeholders and should be updated accordingly.
  - Added template `project_renewal_survey.html`, modeled after `new_project_survey.html`.
  - Imported the form and template into the two views through which a user can request an allocation renewal: `AllocationRenewalRequestUnderProjectView` and `AllocationRenewalRequestView` in `renewal_views/request_views.py`.

**How to Test**

- Manual Testing
  - As a user who is on a project, go to the project's detail page -> _Renew Allowance_, or go to _Project -> Projects -> Renew a PI's Allowance_.
  - Complete the renewal request flow, and ensure the survey is there and has the desired questions depending on if the deployment is LRC or BRC.
  - After submitting the renewal request, go to the Coldfront Administration dashboard and view the newly created Allocation Renewal Request object. Survey answers should appear in JSON format under **Renewal survey answers**.
